modules/src/ics04_channel/context.rs

```diff
@@ -154,6 +154,34 @@ pub trait ChannelKeeper {
                     }
                 }
             }
+
+            PacketResult::Timeout(res) => {
+                match res.channel {
+                    Some(c) => {
+                        //Ordered Channel
+                        let mut channel = c;
```

Comment:
Could we put the `channel` in the result with `.state = State::Closed;`? Then the keeper would just store the `channel`. I'm suggesting this because ideally the keeper should contain minimal IBC logic. It should be the handler typically (not the keeper) that mutates the channel.